test_scripts/AppIconsFolder/ATF_N_AppIconsAmountToRemove_larger_than_icons_amount.lua

```diff
@@ -0,0 +1,245 @@
+    i = i + 1
+    if copyFileToAppIconsFolder ~= true then
+      commonFunctions:userPrint(31, " Files are not copied to " .. tostring(AppIconsFolder))
+    end
+     sizeAppIconsFolderInBytes = folderSize(config.pathToSDL .. tostring(AppIconsFolder))
+     sizeToFull = appIconsFolderMaxSize - sizeAppIconsFolderInBytes
+      if i > 10 then
+        commonFunctions:userPrint(31, " Loop is breaking due to a lot of iterations ")
+        break
+    end
+  end
+end
+
+local function checkFunction()
+  local aHandle = assert( io.popen( "ls " .. config.pathToSDL .. "Icons/" , 'r'))
+  local listOfFilesInStorageFolder = aHandle:read( '*a' )
+  commonFunctions:userPrint(32, "Content of storage folder: " ..tostring("\n" ..listOfFilesInStorageFolder))
+  local iconsFolder = config.pathToSDL .. tostring("Icons/")
+  local applicationFileToCheck = iconsFolder .. RAIParameters.appID
+  local applicationFileExistsResult = checkFileExistence(applicationFileToCheck)
+  if applicationFileExistsResult ~= true then
+    commonFunctions:userPrint(31, tostring(RAIParameters.appID) .. " icon is absent")
+    status = false
+  end
+  for i=1, 3, 1 do
+   local oldFileToCheck = iconsFolder.. "icon" .. tostring(i) ..".png"
+    local oldFileToCheckExistsResult = checkFileExistence(oldFileToCheck)
+    if oldFileToCheckExistsResult~=false then
+      commonFunctions:userPrint(31,"Oldest icon" .. tostring(i).. ".png is not deleted from AppIconsFolder")
+      status = false
+    end
+  end
+    return status
+end
+
+--[[ Preconditions ]]
+commonSteps:DeleteLogsFileAndPolicyTable()
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test.Precondition_stopSDL()
+  StopSDL()
+end 
+
+function Test.Precondition_configureAppIconsFolder()
+  commonFunctions:SetValuesInIniFile("AppIconsFolder%s-=%s-.-%s-\n", "AppIconsFolder", 'Icons')
+end
+ 
+function Test.Precondition_configureAppIconsFolderMaxSize()
+  commonFunctions:SetValuesInIniFile("AppIconsFolderMaxSize%s-=%s-.-%s-\n", "AppIconsFolderMaxSize", 1048576)
+end
+
+function Test.Precondition_configureAppIconsAmountToRemove()
+  commonFunctions:SetValuesInIniFile("AppIconsAmountToRemove%s-=%s-.-%s-\n", "AppIconsAmountToRemove", 100)
+end
+ 
+function Test.Precondition_removeAppIconsFolder()
+  local addedFolderInScript = "Icons"
+    local existsResult = commonSteps:Directory_exist(tostring(config.pathToSDL .. addedFolderInScript))
+  if existsResult == true then
+    local rmAppIconsFolder  = assert( os.execute( "rm -rf " .. tostring(config.pathToSDL .. addedFolderInScript)))
+    if rmAppIconsFolder ~= true then
+      commonFunctions:userPrint(31, tostring(addedFolderInScript) .. " folder is not deleted")
+    end
+  end
+end
+
+function Test.Precondition_startSDL()
+  StartSDL(config.pathToSDL, config.ExitOnCrash)
+end
+
+function Test:Precondition_initHMI()
+  self:initHMI()
+end
+
+function Test:Precondition_initHMIonReady()
+  self:initHMI_onReady()
+end
+
```

Comment:
@aarotar, Following steps can be removed if our 'Test' is inherited from 'user_modules/connecttest_resumption.lua' instead of "connecttest.lua":
- Test.Precondition_startSDL()
- Test:Precondition_initHMI()
- Test:Precondition_initHMIonReady()